src/main/scala/com/databricks/spark/xml/util/InferSchema.scala

```diff
@@ -125,16 +125,7 @@ private[xml] object InferSchema {
       datum
     }
 
-    value match {
-      case null => NullType
-      case v if v.isEmpty => NullType
-      case v if isLong(v) => LongType
-      case v if isInteger(v) => IntegerType
-      case v if isDouble(v) => DoubleType
-      case v if isBoolean(v) => BooleanType
-      case v if isTimestamp(v) => TimestampType
-      case _ => StringType
-    }
+    options.inferStrategy.getDataType(value)
```

Comment:
I think it doesn't need another object. Let's just do an if-else.

Reply:
I was thinking about the same thing.
Since we only have two option for now, if-else would be enough.
Just add the interface to better explain the use case as the first draft solution.
I'm fine for both solution.
Please let me know if modification is required.

Reply:
Let's make it boolean and add an if-else. I have observed the usecases in Spark CSV datasource so far, and I think two cases are good enough.

Reply:
Yeah, I can't think of another use case:
- User specifies schema -> use that
- User asks for inferred schema -> infer types
- Otherwise use the string rep that's already in the XML

That's how CSV works too, so being consistent and using a boolean here may be fine and less surprising.

So, what happens right now if you don't specify a schema? always inferred?

Reply:
Yea, it always infers ..

Reply:
Thanks for organizing the use cases.
I've pushed a version that uses if-else.